text/0005-jsx.md

```diff
@@ -0,0 +1,85 @@
+
+### Working Group
+
+- @jasonLaster
+- ??
+
+## Proposal
+
+Adopt JSX as the default syntax for the `render` method of our react components.
+
+The current DevTools practice for implementing the `render` method of our react components uses the `React.DOM` interface.
+
+Where our current code might look like this:
+
+```js
+import React, { DOM as dom } from "react";
+class X extends React.Component {
+  render() {
+    return dom.div( null,
+      dom.img( { src: "avatar.png", className: "profile" }),
+      dom.h3( null, [user.firstName, user.lastName].join(" "))
+    );
+  }  
+}
+```
+
+A JSX version would look like this:
+
+```js
+import React from "react";
+class X extends React.Component {
+  render() {
+    return (
+      <div>
+        <img src="avatar.png" className="profile" />
+        <h3>{[user.firstName, user.lastName].join(' ')}</h3>
+      </div>
+    );
+  }  
+}
+```
+
+The JSX version is compiled during the build step to JavaScript code.  You can see a live version of the JSX and compiled code [here](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%252Creact%252Cstage-0&targets=&browsers=&builtIns=false&code=import%20React%252C%20%7B%20DOM%20as%20dom%252C%20Component%20%7D%20from%20%22react%22%253B%250Aclass%20X%20extends%20Component%20%7B%250A%20render%28%29%20%7B%250A%20return%20%28%250A%20%3Cdiv%3E%250A%20%3Cimg%20src%253D%22avatar.png%22%20className%253D%22profile%22%20%252F%3E%250A%20%3Ch3%3E%7B%5Buser.firstName%252C%20user.lastName%5D.join%28%22%20%22%29%7D%3C%252Fh3%3E%250A%20%3C%252Fdiv%3E%250A%20%29%253B%250A%20%7D%20%250A%7D) to understand how this above example is compiled into code.
+
+> Fundamentally, JSX just provides syntactic sugar [...](https://facebook.github.io/react/docs/jsx-in-depth.html)
+
+The [JSX](https://facebook.github.io/react/docs/jsx-in-depth.html) page goes into detail on the gotchas related to using JSX. There should not be any DevTools specific issues that arise from this change.
+
+### Alternative solutions
+
+The current `React.DOM` interface is the only other major option available and DevTools is already using this syntax.
+
+### Drawbacks
+
+JSX requires a build step, using something like webpack, so if we feel that a build step is not a requirement we are willing to take on then we should not adopt JSX and continue to use the `React.DOM` practices we currently have.  This build step require can continue to be a per-project decision.
+
+### Adoption
+
+With all new code and as we significantly refactor components of existing projects we should adopt JSX as the standard practice for our `render` methods when there is already a build step required for the project.
+
+### Metrics
+
+Significant metrics will be difficult to gather here. We should see more ownership of the components by less technical contributors and overall a simpler time getting people on boarded.
```

Comment:
Maybe check in after 6 weeks and see if there are places where it doesn't make sense. 

Perhaps look at major components that are JSXd and compare them with components that have not been refactored.